pkg/instance/log.go

```diff
@@ -19,22 +22,51 @@ func (s *Server) LogSetLevel(ctx context.Context, req *rpc.LogSetLevelRequest) (
 	return ops.LogSetLevel(ctx, req)
 }
 
+func logSetLevel(level string) error {
+	// Set instance-manager log level.  We expect a string such as "debug", "info", or "warn".
+	newLevel, err := logrus.ParseLevel(level)
+	if err != nil {
+		return err
+	}
+
+	oldLevel := logrus.GetLevel()
+	if oldLevel != newLevel {
+		logrus.Warnf("Updating log level from %v to %v", oldLevel, newLevel)
+		logrus.SetLevel(newLevel)
+	}
+
+	return nil
+}
+
 func (ops V1DataEngineInstanceOps) LogSetLevel(ctx context.Context, req *rpc.LogSetLevelRequest) (resp *emptypb.Empty, err error) {
-	/* TODO: Implement this */
+	if err := logSetLevel(req.Level); err != nil {
+		return nil, err
+	}
+
 	return &emptypb.Empty{}, nil
 }
 
 func (ops V2DataEngineInstanceOps) LogSetLevel(ctx context.Context, req *rpc.LogSetLevelRequest) (resp *emptypb.Empty, err error) {
+	if err := logSetLevel(req.Level); err != nil {
+		return nil, err
+	}
+
+	// Also set level for spdk_tgt.  There is no "trace" level for SPDK.
+	spdkLevel := strings.ToUpper(req.Level)
+	if spdkLevel == "TRACE" {
+		spdkLevel = "DEBUG"
+	}
```

Comment:
cc @james-munson

Reply:
In general, I agree that hard-coded constants should have a name, to unify all the references and make sure they line up.  These are just one-offs, but I'll make the change, and use names that highlight the point of the check.

For the other (a constant "spdk_tgt", there is no need.  That logic is gone, since there is no req.Component field to check.

Reply:
Done.